engines/dfc_provider/app/controllers/dfc_provider/persons_controller.rb

```diff
@@ -10,6 +10,19 @@ def show
       render json: DfcIo.export(person)
     end
 
+    def affiliate_sales_data
+      render json: DfcIo.export(
+        AffiliateSalesDataBuilder.build_person(user),
+        *AffiliateSalesDataBuilder.build_addresses,
+        *AffiliateSalesDataBuilder.build_producers,
+        *AffiliateSalesDataBuilder.build_supplied_products,
+        *AffiliateSalesDataBuilder.build_catalogue_items,
+        *AffiliateSalesDataBuilder.build_offers,
+        *AffiliateSalesDataBuilder.build_order_lines,
+        *AffiliateSalesDataBuilder.build_orders
+      )
+    end
```

Comment:
@mkllnk, based on what I can see here, it appears that each data type will be presented in one flat array.
But in Garethe's [spec](https://app.swaggerhub.com/apis-docs/RAGGEDSTAFF23/DFC-INRAE-Endpoint/v0.1#/INRAE/get_api_dfc_Persons__id__AffilateSalesData), the related data is nested.

So I'm guessing this will need to be updated to match that format? Or does it not matter, because the receiving software will be able to use the IDs to understand the relationships?

Reply:
The relationships are in the data already. In this `export` call we just list the objects that should be included in the export. If you omit the addresses, for example, then addresses are only referenced by id but not included in the output. So this is all correct.